include/boost/json/basic_parser.hpp

```diff
@@ -2490,6 +2491,10 @@ parse_number(
     }
     else
     {
+        if(BOOST_JSON_UNLIKELY(
+            ! h.on_number_part(
+                {begin, cs.used(begin)}, ec_)))
```

Comment:
@glenfe @pdimov As written, the init-list allows `{begin, cs.used(begin)}` allows `on_number_part` to be declared with an argument type that may not be compatible with string view. Is this desirable? Or should the code above be written as:
```
            ! h.on_number_part(
                string_view(begin, cs.used(begin)),
                ec_)))
```

Reply:
Hard to say. If we require `on_number_part` to take exactly a `string_view` parameter, the above is fine. If we always pass a `string_view`, this allows `on_number_part` to take something convertible from `string_view`, which is probably also undesirable.

Reply:
I haven't bothered to write code to force the handler to have particular signatures (or else get a compile errror) because it doesn't seem worth it.

Reply:
IMO it's worth it because by restricting the signature we can potentially optimize around it.

Reply:
we can restrict the signature without writing any code, just say it in the docs "the signature must match"